model_training/model_training.py

```diff
@@ -0,0 +1,29 @@
+import pickle
+from sklearn import cross_validation
+from sklearn.ensemble import RandomForestClassifier
+
+def train_random_forest(X_train, y_train, estimators, model_name):
+    """
+    Trains a random forest model and stores the trained model as a pickle
+
+    :param X_train: X training data
+    :param y_train: y training data
+    """
+
+    clf = RandomForestClassifier(n_estimators=estimators)
+    clf = clf.fit(X_train, y_train)
+    pickle.dump(clf, open(model_name,'w'))
+
+def split_data(X, y, test_size):
+    """
+    Splits X and y training data into a test and training set
+
+    :param X: all X data
+    :param y: all y data
+    :param test_size: size of the test data set
+    :return: training and test data - X_train, X_test, y_train, y_test
+    """
+
+    return cross_validation.train_test_split(X, y, test_size=test_size, random_state=42)
```

Comment:
Haha, I'm pleased you caught that. It is!

On Mon, Jul 4, 2016 at 1:33 AM, Sean Paradiso <notifications@github.com>
wrote:

> In model_training/model_training.py
> <https://github.com/joanneljhill/image_classifier/pull/1#discussion_r69397720>
> :
>
> > +
> > +    clf = RandomForestClassifier(n_estimators=estimators)
> > +    clf = clf.fit(X_train, y_train)
> > +    pickle.dump(clf, open(model_name,'w'))
> > +
> > +def split_data(X, y, test_size):
> > +    """
> > +    Splits X and y training data into a test and training set
> > +
> > +    :param X: all X data
> > +    :param y: all y data
> > +    :param test_size: size of the test data set
> > +    :return: training and test data - X_train, X_test, y_train, y_test
> > +    """
> > +
> > +    return cross_validation.train_test_split(X, y, test_size=test_size, random_state=42)
>
> please tell me this is a hitchhiker's guide reference!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/joanneljhill/image_classifier/pull/1/files/088c87f529f12e6c3d8babe84ab5e718c058e13b..2642d6deeae0f3952b35bf1a4d7d428df8acc8b1#r69397720>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ANKctas-yH1tPA6faO5y_aPEeZm9VQJLks5qSCqrgaJpZM4JBbml>
> .
>